rclpy/rclpy/parameter_event_handler.py

```diff
@@ -0,0 +1,367 @@
+    def get_parameter_from_event(
+        event: ParameterEvent,
+        parameter_name: str,
+        node_name: str,
+    ) -> Optional[Parameter]:
+        """
+        Get specified parameter value from ParameterEvent message.
+
+        :param event: ParameterEvent message to be read
+        :param parameter_name: Name of a parameter to get from ParameterEvent message
+        :param node_name: Name of a node, that the parameter should be related to
+
+        :return Optional[Parameter]: If specified parameter is found, returns Parameter object.
+                                     Otherwise, returns None
+        """
+        if event.node == node_name:
+            for parameter in chain(event.new_parameters, event.changed_parameters):
+                if parameter.name == parameter_name:
+                    return parameter
+
+        return None
+
+    @staticmethod
+    def get_parameters_from_event(event: ParameterEvent) -> Iterable[Parameter]:
+        """
+        Get all parameters from a ParameterEvent message.
+
+        :param event: ParameterEvent message to read
+        """
+        for parameter in chain(event.new_parameters, event.changed_parameters):
+            yield parameter
+
+    def add_parameter_callback(
+        self,
+        parameter_name: str,
+        node_name: str,
+        callback: Callable[[Parameter], None],
+    ) -> ParameterCallbackHandle:
+        """
+        Add new parameter callback.
+
+        Callbacks are called in FILO manner.
+
+        :param parameter_name: Name of a parameter to tie callback to
+        :param node_name: Name of a node, that the parameter should be related to
+        :param callback: A callable to be referenced on a parameter permutation
+
+        :return ParameterCallbackHandle: The duplicate of a callback handle,
+                                         stored in a parameter_event dict
+        """
+        return self._callbacks.add_parameter_callback(
+            parameter_name=parameter_name,
+            node_name=self.resolve_path(node_name),
+            callback=callback,
+        )
+
+    def remove_parameter_callback(
+        self,
+        handle: ParameterCallbackHandle,
+    ) -> None:
+        """
+        Remove a ParameterCallbackHandle.
+
+        Perform no callbacks on this parameter events in the future.
+
+        :param handle: ParameterCallbackHandle of a callback t obe removed
+        """
+        self._callbacks.remove_parameter_callback(handle)
+
+    def add_parameter_event_callback(
+        self,
+        callback: Callable[[ParameterEvent], None],
+    ) -> ParameterEventCallbackHandle:
+        """
+        Add new parameter callback.
+
+        Callbacks are called in FILO manner.
+
+        :param callback: A callable to be referenced on a ParameterEvent
+
+        :return ParameterEventCallbackHandle: The duplicate of a callback handle,
+                                              stored in a parameter_event dict
```

Comment:
I'm not sure the information in this is that useful to a user of this class.  How about something like:

```suggestion
        :return ParameterEventCallbackHandle: A handle that should be saved by the user so that the event_callback can be later removed.
```